internal/bkmultipart/streamer_test.go

```diff
@@ -0,0 +1,114 @@
+package bkmultipart_test
+
+import (
+	"bytes"
+	"io"
+	"mime"
+	"mime/multipart"
+	"os"
+	"strings"
+	"testing"
+
+	"github.com/buildkite/go-buildkite/v3/internal/bkmultipart"
+)
+
+func TestEncodingFormFields(t *testing.T) {
+	t.Parallel()
+
+	s := bkmultipart.NewStreamer()
+	fields := map[string]string{
+		"mountain": "cotopaxi",
+		"city":     "guayaquil",
+	}
+	if err := s.WriteFields(fields); err != nil {
+		t.Fatalf("WriteFields(%# v) = %v, want nil", fields, err)
+	}
+
+	form := parseForm(t, s)
+	for key, values := range form.Value {
+		if len(values) != 1 {
+			t.Fatalf("got %d values for %s, want 1", len(values), key)
+		}
+		if got, want := values[0], fields[key]; got != want {
+			t.Fatalf("form.Value[%q] = [%q], want [%q]", key, got, want)
+		}
+	}
+}
+
+func TestEncodingFile(t *testing.T) {
+	t.Parallel()
+
+	tempFile, err := os.CreateTemp("", "buildkite-go-sdk-test")
+	if err != nil {
+		t.Fatalf("os.CreateTemp() = %v, want nil", err)
+	}
+
+	t.Cleanup(func() { os.Remove(tempFile.Name()) })
+	t.Cleanup(func() { tempFile.Close() })
```

Comment:
Since you need to wrap the calls in a func anyway, you could do this (making the order of operations clearer):
```suggestion
	t.Cleanup(func() {
		tempFile.Close()
		os.Remove(tempFile.Name())
	})
```

Reply:
ooh yes, good call